app/src/main/java/com/github/se/studentconnect/model/notification/NotificationRepositoryFirestore.kt

```
@@ -0,0 +1,197 @@
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  override fun markAsRead(
+      notificationId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(NOTIFICATIONS_COLLECTION)
+        .document(notificationId)
+        .update("isRead", true)
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  override fun markAllAsRead(
+      userId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(NOTIFICATIONS_COLLECTION)
+        .whereEqualTo("userId", userId)
+        .whereEqualTo("isRead", false)
+        .get()
+        .addOnSuccessListener { snapshot ->
+          val batch = db.batch()
+          snapshot.documents.forEach { doc -> batch.update(doc.reference, "isRead", true) }
+          batch
+              .commit()
+              .addOnSuccessListener { onSuccess() }
+              .addOnFailureListener { exception -> onFailure(exception) }
+        }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  override fun deleteNotification(
+      notificationId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(NOTIFICATIONS_COLLECTION)
+        .document(notificationId)
+        .delete()
+        .addOnSuccessListener { onSuccess() }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
+
+  override fun deleteAllNotifications(
+      userId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(NOTIFICATIONS_COLLECTION)
+        .whereEqualTo("userId", userId)
+        .get()
+        .addOnSuccessListener { snapshot ->
+          val batch = db.batch()
+          snapshot.documents.forEach { doc -> batch.delete(doc.reference) }
+          batch
+              .commit()
+              .addOnSuccessListener { onSuccess() }
```

Same issue as markAllAsRead - Firestore batch limit of 500 operations not handled. This will fail silently for users with many notifications.

---

fixed! applied the same chunking logic here, operations are now split into batches of 500 to respect Firestore's limit.